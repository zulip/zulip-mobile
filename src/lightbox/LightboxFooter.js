/* @flow */
import React from 'react';
import { Text, StyleSheet, View } from 'react-native';

import type { StyleObj } from '../types';
import { SlideAnimationView } from '../common';
import NavButton from '../nav/NavButton';

const styles = StyleSheet.create({
  text: {
    color: 'white',
    fontSize: 16,
    lineHeight: 16,
  },
  icon: {
    fontSize: 32,
  },
});

type FooterProps = {
  displayMessage: string,
  onOptionsPress: () => void,
  style: StyleObj,
};

type Props = {
  displayMessage: string,
  onOptionsPress: () => void,
  from: number,
  to: number,
  style: StyleObj,
};

export const Footer = ({ displayMessage, onOptionsPress, style }: FooterProps) =>
  <View style={style}>
    <Text style={styles.text}>
      {displayMessage}
    </Text>
    <NavButton name="ios-more" color="white" style={styles.icon} onPress={onOptionsPress} />
  </View>;

export default ({ displayMessage, onOptionsPress, style, ...restProps }: Props) =>
  <SlideAnimationView property={'translateY'} style={style} {...restProps}>
    <Footer displayMessage={displayMessage} onOptionsPress={onOptionsPress} style={style} />
  </SlideAnimationView>;
