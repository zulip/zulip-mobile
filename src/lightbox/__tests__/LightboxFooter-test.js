import React from 'react';
import renderer from 'react-test-renderer';

import { Footer } from '../LightboxFooter';

describe('LightboxFooter', () => {
  test('all required props are passed correctly', () => {
    const displayMessage = 'Shared in #mobile';
    const onOptionsPress = jest.fn();

    const footer = renderer.create(
      <Footer displayMessage={displayMessage} onOptionsPress={onOptionsPress} style={{}} />,
    );

    expect(footer).toMatchSnapshot();
  });
});
