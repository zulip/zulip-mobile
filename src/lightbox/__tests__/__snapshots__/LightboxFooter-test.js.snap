// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LightboxFooter all required props are passed correctly 1`] = `
<View
  style={Object {}}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 16,
        "lineHeight": 16,
      }
    }
  >
    Shared in #mobile
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 44,
            "justifyContent": "center",
            "width": 44,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 12,
              },
              Array [
                Object {
                  "fontSize": 26,
                  "textAlign": "center",
                },
                Object {
                  "fontSize": 32,
                },
              ],
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          ï‡ˆ
        </Text>
      </View>
    </View>
  </View>
</View>
`;
